components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIProvider.java

```diff
@@ -1839,4 +1838,163 @@ EnvironmentPropertiesDTO getEnvironmentSpecificAPIProperties(String apiUuid, Str
      * @return List of environments related to the given tenant
      */
     Environment getEnvironment(String organization, String uuid) throws APIManagementException;
+
+    /**
+     * Set existing operation policy mapping to the URI Templates
+     *
+     * @param apiId        API UUID
+     * @param uriTemplates Set of URI Templates
+     * @throws APIManagementException
+     */
+    void setOperationPoliciesToURITemplates(String apiId, Set<URITemplate> uriTemplates) throws APIManagementException;
+
+    /**
+     * Import an operation policy from the API CTL project. This will either create a new API specific policy,
+     * update existing API specific policy or return the policyID of existing policy if policy content is not changed.
+     *
+     * @param operationPolicyData Operation Policy Data
+     * @param organization              Organization name
+     * @return UUID of the imported operation policy
+     * @throws APIManagementException
+     */
+    String importOperationPolicy(OperationPolicyData operationPolicyData, String organization)
```

Comment:
do we need this since we already have addCommonOperationPolicy

Reply:
Seems like there is a usage for `importOperationPolicy`, and it revolves around the CTL. It is used to import an operation policy from the apictl project. I think we need this. @1akshitha please correct me if I'm mistaken.